helpdesk_mgmt/models/helpdesk_ticket.py

```diff
@@ -146,6 +146,12 @@ def create(self, vals_list):
                 vals["number"] = self._prepare_ticket_number(vals)
             if vals.get("user_id") and not vals.get("assigned_date"):
                 vals["assigned_date"] = fields.Datetime.now()
+            if vals.get("team_id"):
+                vals["company_id"] = (
+                    self.env["helpdesk.ticket.team"]
+                    .browse([vals["team_id"]])
+                    .company_id.id
+                )
```

Comment:
```suggestion
                team = self.env["helpdesk.ticket.team"].browse([vals["team_id"]])
                if team.company_id:
                    vals["company_id"] = team.company_id.id
```
Similar to https://github.com/OCA/helpdesk/pull/604